pysolarmanv5/pysolarmanv5.py

```diff
@@ -284,81 +278,89 @@ def _send_receive_v5_frame(self, data_logging_stick_frame):
             if v5_response == b"":
                 raise NoSocketAvailableError("Connection closed on read")
             self._data_wanted.clear()
-        except (queue.Empty, TimeoutError) as e:
-            self.log.debug(f"Got exception when receiving frame: {e}")
+        except (queue.Empty, TimeoutError):
+            self.log.debug("Got exception when receiving frame", exc_info=True)
             raise
 
-        self.log.debug("RECD: " + v5_response.hex(" "))
+        self.log.debug("RECD: %s", v5_response.hex(" "))
         return v5_response
 
+    def _received_frame_is_valid(self, frame):
+        """Check that the frame is valid and that the serial number of the received
+        frame matches with the last sent one.
+        Ignore also any counter frames.
+        """
+        if frame.startswith(b"\xa5\x01\x00\x10G"):
+            # Frame with control code 0x4710 - Counter frame
+            self.log.debug("[{self.serial}] COUNTER: %s", frame.hex(" "))
```

Comment:
`{self.serial}` should be problematic in this block

Reply:
Indeed... will fix